tmplctlr/controller_test.go

```diff
@@ -64,11 +64,13 @@ func createTestDir(files []templateFile) string {
 		if err != nil {
 			log.Fatal(err)
 		}
-		defer f.Close()
-		_, err = io.WriteString(f, file.contents)
-		if err != nil {
-			log.Fatal(err)
-		}
+		func() {
+			defer f.Close()
+			_, err = io.WriteString(f, file.contents)
+			if err != nil {
+				log.Fatal(err)
+			}
+		}()
```

Comment:
This addresses a warning about using defer in a loop. Forcing defer to happen at the end of each anonymous function call